packages/opentelemetry-metrics/src/MeterProvider.ts

```diff
@@ -14,27 +14,53 @@
  * limitations under the License.
  */
 
-import { ConsoleLogger } from '@opentelemetry/core';
 import * as api from '@opentelemetry/api';
+import { ExportResult, ConsoleLogger } from '@opentelemetry/core';
+import { Meter } from './Meter';
+import { MetricExporter } from './export/types';
+import { Batcher, UngroupedBatcher } from './export/Batcher';
 import { Resource } from '@opentelemetry/resources';
-import { Meter } from '.';
-import { DEFAULT_CONFIG, MeterConfig } from './types';
+import { MeterConfig, DEFAULT_CONFIG } from './types';
+import { NoopExporter } from './export/NoopExporter';
 
 /**
  * This class represents a meter provider which platform libraries can extend
  */
 export class MeterProvider implements api.MeterProvider {
   private readonly _config: MeterConfig;
   private readonly _meters: Map<string, Meter> = new Map();
+  private readonly _batcher: Batcher;
+  private readonly _exporter: MetricExporter;
   readonly resource: Resource;
   readonly logger: api.Logger;
 
   constructor(config: MeterConfig = DEFAULT_CONFIG) {
     this.logger = config.logger ?? new ConsoleLogger(config.logLevel);
     this.resource = config.resource ?? Resource.createTelemetrySDKResource();
+    this._batcher = config.batcher ?? new UngroupedBatcher();
+    this._exporter = config.exporter ?? new NoopExporter();
     this._config = Object.assign({}, config, {
       logger: this.logger,
       resource: this.resource,
+      batcher: this._batcher,
+    });
+  }
+
+  async collect() {
+    await Promise.all(
+      Array.from(this._meters.values()).map(meter => meter.collect())
+    );
+    return new Promise(resolve => {
+      this._exporter.export(this._batcher.checkPointSet(), result => {
+        // TODO: retry strategy
+        if (result !== ExportResult.SUCCESS) {
+          this.logger.error(
+            'Metric exporter reported non success result(%s).',
+            result
+          );
+        }
+        resolve();
```

Comment:
Curious why only resolve without reject ?

Reply:
Moved the export result handling to the push controller so that we can implement retry with controllers.